cli/src/commands/git/fetch.rs

```diff
@@ -41,8 +41,9 @@ pub struct GitFetchArgs {
     ///
     /// By default, the specified name matches exactly. Use `glob:` prefix to
     /// expand `*` as a glob. The other wildcard characters aren't supported.
-    #[arg(long, short, default_value = "glob:*", value_parser = StringPattern::parse)]
-    branch: Vec<StringPattern>,
+    // TODO: Remove the branch alias in jj 0.27+
```

Comment:
Similarly to my other comment, https://github.com/martinvonz/jj/pull/4341/files#r1744195081, should we keep this as `branch`? Should we just explain that we're fetching remote branches to get local bookmarks?

Same question exists for `jj git push`.

Reply:
I think we fetch _remote branches_ as local bookmarks and we push _local bookmarks_ as Git branches.

Reply:
I think we said the same thing.

I think we should explain something about this in the docstring for `git push` and `git fetch` (again, not sure whether that should be in this PR or a separate one). Moreover, I'm tempted to keep both `branch` and `bookmark` args long-term, though I'm not completely sure about it.

Reply:
> Moreover, I'm tempted to keep both `branch` and `bookmark` args long-term, though I'm not completely sure about it.

I'm fine with keeping `branch` as an argument long-term for `jj git fetch`, as we're fetching a Git branches. With Topics they hopefully won't map to bookmarks though.

Reply:
Conceptually, it'd make a lot of sense to have `jj git fetch --branch` and `jj git push --bookmark`. I'm worried it might be confusing, though, so perhaps both commands should accept both?

> With Topics they hopefully won't map to bookmarks though.

This makes me lean harder towards using `--branch` here.

-------

I'm still not 100% certain what's best here, so if this discussion is at some point the only blocker for the PR, we can probably merge the PR and wonder about it later. But if we come to a consensus, even better.

Reply:
I've added a sentence here, is that ok for you?

Reply:
> Conceptually, it'd make a lot of sense to have `jj git fetch --branch` and `jj git push --bookmark`. I'm worried it might be confusing, though, so perhaps both commands should accept both?

SGTM, although I think that `fetch` should only provide `--branch` while with `git push --bookmark` makes more sense for a bookmark, for a later change to support something like `git push --topic`

Reply:
I've reverted the change to `fetch` now, so it keeps its `--branch` argument.